src/storage/knn_index/knn_diskann/knn_diskann.cppm

```diff
@@ -0,0 +1,124 @@
+// Copyright(C) 2023 InfiniFlow, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+module;
+
+export module knn_diskann;
+
+import stl;
+import knn_distance;
+
+import infinity_exception;
+import index_base;
+import kmeans_partition;
+import search_top_1;
+import search_top_k;
+import knn_result_handler;
+import bitmask;
+import knn_expr;
+import internal_types;
+import logger;
+import diskann_index_data;
+import diskann_dist_func;
+
+namespace infinity {
+template <typename Compare, MetricType metric, KnnDistanceAlgoType algo>
+class KnnDiskAnn final : public KnnDistance<typename Compare::DistanceType> {
```

Comment:
You have two class template named "KnnDiskAnn", another is in "vamana_alg.cppm", which is confusing.